Lib/asyncio/tasks.py

```diff
@@ -334,6 +334,11 @@ def wait_for(fut, timeout, *, loop=None):
     if timeout is None:
         return (yield from fut)
 
+    if timeout == 0:
```

Comment:
You probably want to put `timeout <= 0` here to enable this behaviour for timeouts that were computed.